tools/data-handler/src/utils/clingo-facts.ts

```diff
@@ -0,0 +1,260 @@
+import { sep } from 'path';
+import { Card } from '../interfaces/project-interfaces.js';
+import {
+  CardType,
+  FieldType,
+  Link,
+  LinkType,
+  Workflow,
+} from '../interfaces/resource-interfaces.js';
+import { ClingoProgramBuilder } from './clingo-program-builder.js';
+import { ClingoFactBuilder } from './clingo-fact-builder.js';
+import { Project } from '../containers/project.js';
+
+export const createWorkflowFacts = (workflow: Workflow) => {
+  const builder = new ClingoProgramBuilder().addFact('workflow', workflow.name);
+  // add states
+  for (const state of workflow.states) {
+    if (state.category) {
+      builder.addFact(
+        'workflowState',
+        workflow.name,
+        state.name,
+        state.category,
+      );
+    } else {
+      builder.addFact('workflowState', workflow.name, state.name);
+    }
+  }
+
+  // add transitions
+  for (const transition of workflow.transitions) {
+    for (const from of transition.fromState) {
+      builder.addFact(
+        'workflowTransition',
+        workflow.name,
+        transition.name,
+        from,
+        transition.toState,
+      );
+    }
+    if (transition.fromState.length === 0) {
+      builder.addFact(
+        'workflowTransition',
+        workflow.name,
+        transition.name,
+        '',
+        transition.toState,
+      );
+    }
+  }
+  return builder.buildAll();
+};
+
+export const createCardFacts = async (card: Card, project: Project) => {
+  // Small helper to deduce parent path
+  function parentPath(cardPath: string) {
```

Comment:
No need to do in this PR, but can you make a tech debt ticket that we should make this a public generic function?

Reply:
Yup